### PR TITLE
5.x: add options to enable gpu in containers

### PR DESCRIPTION
**Pull request by fengyuentau**
Relates https://github.com/opencv/opencv/pull/26302

### CODE REVIEW

File: .github/workflows/OCV-PR-5.x-ARM64.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ jobs:
         shell: bash
     container:
       image: quay.io/opencv-ci/opencv-ubuntu-20.04-arm64:20230413
+      options: --gpus all -m 8G
```
Comment (asmorkalov) [inline .github/workflows/OCV-PR-5.x-ARM64.yaml:38]: Looks like --gpu option is not supported on ARM here and above:
```
Error response from daemon: could not select device driver "" with capabilities: [[gpu]]
```

**Review by asmorkalov — CHANGES_REQUESTED** [review]

File: .github/workflows/OCV-PR-5.x-ARM64.yaml (hunk 1 above)
Comment (asmorkalov) [reply .github/workflows/OCV-PR-5.x-ARM64.yaml:38]: Please ignore audio test failure. It's know sporadic issue related to sound.

Comment (fengyuentau) [reply .github/workflows/OCV-PR-5.x-ARM64.yaml:38]: I just checked available `display` or `VGA` devices on the ARM host and it turns out that the chip is not integrated with a GPU that supports OpenCL. We can set `-DWITH_OPENCL=OFF` for the ARM host.

**Comment by fengyuentau** [comment]
@asmorkalov Please review again. I will merge if no more comments.

**Review by asmorkalov — APPROVED** [review]

File: .github/workflows/OCV-PR-5.x-ARM64.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  EXTRA_CMAKE_OPTIONS: '-DBUILD_DOCS=ON -DPYTHON_DEFAULT_EXECUTABLE=/usr/bin/python3 -DOPENCV_DOWNLOAD_PATH=/home/ci/binaries_cache -DBUILD_EXAMPLES=ON -DOPENCV_ENABLE_NONFREE=ON'
+  EXTRA_CMAKE_OPTIONS: '-DBUILD_DOCS=ON -DPYTHON_DEFAULT_EXECUTABLE=/usr/bin/python3 -DOPENCV_DOWNLOAD_PATH=/home/ci/binaries_cache -DBUILD_EXAMPLES=ON -DOPENCV_ENABLE_NONFREE=ON -DWITH_OPENCL=OFF'
```
Comment (opencv-alalek) [inline .github/workflows/OCV-PR-5.x-ARM64.yaml:16]: `WITH_OPENCL` is `ON` by default.

We should check that this "default" mode doesn't crash even without OpenCL devices or any OpenCL runtime.

`WITH_OPENCL=OFF` makes sense for dedicated debug ""no-optimization"  builds only.

Comment (fengyuentau) [reply .github/workflows/OCV-PR-5.x-ARM64.yaml:16]: >  this "default" mode doesn't crash even without OpenCL devices or any OpenCL runtime.

Sometimes it causes accuracy problem if we do some changes on the CPU code (like in my warp functions optimization PRs). The changes are all about CPU side, but the OCL tests failed and needs to be changed even though OpenCL code is not touched at all.

**Pull request merged**